R/zchunk_L261.Cstorage.R

```diff
@@ -56,204 +68,319 @@ module_energy_L261.Cstorage_DISABLED <- function(command, ...) {
+                           GCAM_region_names = GCAM_region_names) ->
+      L261.SubsectorShrwtFllt_C # This is a final output table.
+
+
+    # E
+    # Technology information
+    # Identification of stub technologies of carbon storage
+    # A61.globaltech_shrwt reports carbon storage technology shareweights
+    # Note: assuming that the technology list in the shareweight table includes the full set (any others would default to a 0 shareweight)
+    A61.globaltech_shrwt %>%
+      write_to_all_regions(LEVEL2_DATA_NAMES[["Tech"]],
+                           GCAM_region_names = GCAM_region_names) %>%
+      select(region, supplysector, subsector, stub.technology = technology) ->
+      L261.StubTech_C # This is a final output table.
+
+    # Energy inputs and coefficients of global technologies for carbon storage
+    # A61.globaltech_coef reports carbon storage global technology coefficients
+    A61.globaltech_coef %>%
+      gather(year, value, matches(YEAR_PATTERN)) %>% # Convert to long form
+      mutate(year = as.integer(year), # Year needs to be integer or numeric to interpolate
+             value = as.numeric(value)) %>%
+      # Expand table to include all model base and future years
+      group_by(supplysector, subsector, technology, minicam.energy.input) %>%
+      complete(year = c(year, MODEL_YEARS)) %>%
+      # Extrapolate to fill out values for all years
+      # Rule 2 is used so years outside of min-max range are assigned values from closest data, as opposed to NAs
+      mutate(value = approx_fun(year, value, rule = 2)) %>%
+      ungroup() %>%
+      mutate(year = as.integer(year)) %>% # Year was converted to numeric after interpolation operation. Convert back to integer.
+      filter(year %in% MODEL_YEARS) %>% # This will drop 1971
+      # Assign the columns "sector.name" and "subsector.name", consistent with the location info of a global technology
+      select(sector.name = supplysector, subsector.name = subsector, technology, year, minicam.energy.input, coefficient = value) ->
+      L261.GlobalTechCoef_C # This is a final output table.
+
+    # Costs of global technologies
+    # A61.globaltech_cost reports carbon storage offshore storage cost (1975$/tCO2)
+    A61.globaltech_cost %>%
+      gather(year, value, matches(YEAR_PATTERN)) %>%
+      mutate(year = as.integer(year),
+             value = as.numeric(value)) %>%
+      # Expand table to include all model base and future years
+      group_by(supplysector, subsector, technology, minicam.non.energy.input) %>%
+      complete(year = c(year, MODEL_YEARS)) %>%
+      # Extrapolate to fill out values for all years
+      # Rule 2 is used so years outside of min-max range are assigned values from closest data, as opposed to NAs
+      mutate(value = approx_fun(year, value, rule = 2)) %>%
+      ungroup() %>%
+      mutate(year = as.integer(year)) %>% # Year was converted to numeric after interpolation operation. Convert back to integer.
```

Comment:
Definitely don't leave as character; but I don't think it matters if it's numeric vs. integer.